sycl/CMakeLists.txt

```diff
@@ -228,15 +228,10 @@ if (SYCL_ENABLE_XPTI_TRACING)
   endif()
 endif()
 
-if (NOT DEFINED LLVM_INCLUDE_TESTS)
-  set(LLVM_INCLUDE_TESTS ON)
```

Comment:
Let's discuss if we need this code or if we can replace it with something better.
This snippet was added by https://github.com/intel/llvm/pull/585 and it seems to handle the case when SYCL runtime project is built separately from llvm. I don't think we currently support this case, so it can be safely removed.
If we need to support this case we should:
1. add regular testing for this configuration
1. handle it more accurately - e.g. like libcxx does [here](https://github.com/intel/llvm/blob/sycl/libcxx/cmake/Modules/HandleOutOfTreeLLVM.cmake)

Reply:
Agree, to remove. @bjoernknafla Do you see any reason why we should keep it?

Reply:
I have [mirrored how Clang does it](https://github.com/intel/llvm/blob/sycl/clang/CMakeLists.txt#L121) but am not sure if Clang is a special case compared to the other LLVM projects that deal differently with it, e.g., LLD.

As long as we ensure that `SYCL_INCLUDE_TESTS` is set in the buildbots ([as it currently is](https://github.com/intel/llvm/blob/sycl/buildbot/configure.py#L83`0) I see no reason to remove it though.

It might surprise users though that suddenly the tests are off, and they might see tests pass because they aren’t tests, if they haven’t explicitly turned the SYCL ones on?

Reply:
According to my understanding, as long as SYCL project is build withing LLVM tree, we don't need to set `SYCL_INCLUDE_TESTS` explicitly.
I've tried it locally by removing [this line](https://github.com/intel/llvm/blob/sycl/buildbot/configure.py#L83), and test target is included.
This code handles the case if LLVM CMake files are not processed before SYCL CMake files and `LLVM_INCLUDE_TESTS` variable is not set. As I mentioned in the comments above, this configuration is not supported and we have a lot of other variables to set in addition to `LLVM_INCLUDE_TESTS` to use SYCL project in stand-alone mode.

@bjoernknafla, are you building SYCL project separately from LLVM?

Reply:
@bader I am not building SYCL separately from LLVM. In the future it would be helpful to enable separate building as this could speed up testing, e.g., a compiler change needs to build and run all tests, a runtime change only needs to build and run runtime tests - just a thought and nothing that could inform a variable change right now.

Currently `SYCL_INCLUDE_TESTS` is set automatically to `LLVM_INCLUDE_TESTS`: [sycl/CMakeLists.txt#L244](https://github.com/intel/llvm/blob/sycl/sycl/CMakeLists.txt#L244)

How did you test after the CMake configuration change, e.g., did you use the buildbot scripts (which set the variable implicitly), and did you delete the CMake cache and run the configuration freshly? (Only asking to prevent turning tests off by accident which has bitten me in the past).

Reply:
> Currently `SYCL_INCLUDE_TESTS` is set automatically to `LLVM_INCLUDE_TESTS`: [sycl/CMakeLists.txt#L244](https://github.com/intel/llvm/blob/sycl/sycl/CMakeLists.txt#L244)

`LLVM_INCLUDE_TESTS` default value is `ON`, so if sycl project is build within llvm tree, cmake parses LLVM project cmake files and set `LLVM_INCLUDE_TESTS` variable to `ON` (unless it's explicitly set to `OFF` by user. This means that `SYCL_INCLUDE_TESTS` default value is `ON` in our current configuration.
 
> How did you test after the CMake configuration change, e.g., did you use the buildbot scripts (which set the variable implicitly), and did you delete the CMake cache and run the configuration freshly? (Only asking to prevent turning tests off by accident which has bitten me in the past).

I removed the whole build directly and removed the line 83 from the configure.py script. Then I run the script to generate Ninja files and run `ninja check-sycl` command, which works as expected.

Reply:
Thank you for helping me see your full picture. Sounds good to me.